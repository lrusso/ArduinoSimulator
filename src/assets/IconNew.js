import React from "react"

const IconNew = ({ width, height }) => {
  return (
    <svg width={width} height={height} viewBox="0 0 1000 1000">
      <path
        d="M903 761l-142 142q-24 24 -64 40.5t-73 16.5h-480q-33 0 -56.5 -23.5t-23.5 -56.5v-864q0 -33 23.5 -56.5t56.5 -23.5h736q33 0 56.5 23.5t23.5 56.5v608q0 33 -16.5 73t-40.5 64zM858 715q3 -2 5 -5l4 -6h-163v163l6 -4t5 -5zM896 16q0 -7 -4.5 -11.5t-11.5 -4.5h-736
    q-7 0 -11.5 4.5t-4.5 11.5v864q0 7 4.5 11.5t11.5 4.5h480q4 0 8 -0.5t8 -1.5v-254h254q1 -4 1.5 -8t0.5 -8v-608z"
        transform="translate(0 930) scale(-1,1) rotate(180)"
      />
    </svg>
  )
}

export default IconNew
