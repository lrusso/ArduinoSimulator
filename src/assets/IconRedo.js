import React from "react"

const IconRedo = ({ width, height }) => {
  return (
    <svg width={width} height={height} viewBox="0 0 1000 1000">
      <path
        d="M576 712v248l384 -384l-384 -384v254q-168 4 -253.5 -46.5t-113.5 -127.5t-9 -168t62 -168q-105 114 -144.5 251.5t-4.5 255.5t148.5 196t314.5 73v0z"
        transform="translate(0 930) scale(-1,1) rotate(180)"
      />
    </svg>
  )
}

export default IconRedo
