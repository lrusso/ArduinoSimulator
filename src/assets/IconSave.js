import React from "react"

const IconSave = ({ width, height }) => {
  return (
    <svg width={width} height={height} viewBox="0 0 1000 1000">
      <path
        d="M896 960h-896v-1024h1024v896zM512 832h128v-256h-128v256zM896 64h-768v768h64v-320h576v320h75l53 -53v-715z"
        transform="translate(-10 930) scale(-1,1) rotate(180)"
      />
    </svg>
  )
}

export default IconSave
