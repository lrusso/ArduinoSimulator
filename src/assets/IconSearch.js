import React from "react"

const IconSearch = ({ width, height }) => {
  return (
    <svg width={width} height={height} viewBox="0 0 24 24">
      <path
        d="M11,5 C8.24078475,5 6,7.24078475 6,10 C6,12.7592153 8.24078475,15 11,15 C13.7592153,15 16,12.7592153 16,10 C16,7.24078475 13.7592153,5 11,5 Z M16.1517226,14.737509 L19.7071068,18.2928932 C20.0976311,18.6834175 20.0976311,19.3165825 19.7071068,19.7071068 C19.3165825,20.0976311 18.6834175,20.0976311 18.2928932,19.7071068 L14.5932864,16.0075 C13.5426554,16.6376337 12.3133603,17 11,17 C7.13621525,17 4,13.8637847 4,10 C4,6.13621525 7.13621525,3 11,3 C14.8637847,3 18,6.13621525 18,10 C18,11.826606 17.2990779,13.4906052 16.1517226,14.737509 Z"
        transform="translate(0 2)"
      />
    </svg>
  )
}

export default IconSearch
