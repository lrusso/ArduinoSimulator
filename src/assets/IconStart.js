import React from "react"

const IconStart = ({ width, height }) => {
  return (
    <svg
      width={width}
      height={height}
      viewBox="0 0 56.693 56.693"
      id="arduino_simulator_start_icon"
    >
      <path d="M46.279,26.484L20.113,7.48c-0.957-0.695-2.228-0.796-3.278-0.258c-1.056,0.537-1.721,1.622-1.721,2.807  v38.009c0,1.184,0.665,2.268,1.721,2.805c0.45,0.23,0.939,0.344,1.429,0.344c0.652,0,1.303-0.203,1.85-0.602l26.166-19.006  c0.818-0.594,1.298-1.539,1.298-2.548C47.577,28.023,47.094,27.077,46.279,26.484z" />
    </svg>
  )
}

export default IconStart
