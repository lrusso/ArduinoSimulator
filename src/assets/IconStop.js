import React from "react"

const IconStop = ({ width, height }) => {
  return (
    <svg
      width={width}
      height={height}
      viewBox="0 0 24 24"
      id="arduino_simulator_stop_icon"
    >
      <path
        d="M16,6H8C6.9,6,6,6.9,6,8v8c0,1.1,0.9,2,2,2h8c1.1,0,2-0.9,2-2V8C18,6.9,17.1,6,16,6z"
        transform="translate(-0.5 -5) scale(1.1,1.5)"
      />
    </svg>
  )
}

export default IconStop
