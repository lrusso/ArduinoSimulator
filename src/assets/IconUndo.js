import React from "react"

const IconUndo = ({ width, height }) => {
  return (
    <svg width={width} height={height} viewBox="0 0 1000 1000">
      <path
        d="M762 -64q43 77 62 168t-9 168t-113.5 127.5t-253.5 46.5v-254l-384 384l384 384v-248q201 5 314.5 -73t148.5 -196t-4.5 -255.5t-144.5 -251.5z"
        transform="translate(0 930) scale(-1,1) rotate(180)"
      />
    </svg>
  )
}

export default IconUndo
