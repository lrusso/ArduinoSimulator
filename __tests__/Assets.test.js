import React from "react"
import renderer from "react-test-renderer"
import IconBoard from "../src/assets/IconBoard"
import IconNew from "../src/assets/IconNew"
import IconOpen from "../src/assets/IconOpen"
import IconRedo from "../src/assets/IconRedo"
import IconStart from "../src/assets/IconStart"
import IconSave from "../src/assets/IconSave"
import IconSearch from "../src/assets/IconSearch"
import IconStop from "../src/assets/IconStop"
import IconUndo from "../src/assets/IconUndo"

describe("Assets", () => {
  test("IconBoard Enabled", () => {
    const tree = renderer
      .create(<IconBoard width={16} height={16} enabled />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  test("IconBoard Disabled", () => {
    const tree = renderer.create(<IconBoard width={16} height={16} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  test("IconNew", () => {
    const tree = renderer.create(<IconNew width={16} height={16} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  test("IconOpen", () => {
    const tree = renderer.create(<IconOpen width={16} height={16} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  test("IconRedo", () => {
    const tree = renderer.create(<IconRedo width={16} height={16} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  test("IconStart", () => {
    const tree = renderer.create(<IconStart width={16} height={16} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  test("IconSave", () => {
    const tree = renderer.create(<IconSave width={16} height={16} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  test("IconSearch", () => {
    const tree = renderer.create(<IconSearch width={16} height={16} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  test("IconStop", () => {
    const tree = renderer.create(<IconStop width={16} height={16} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  test("IconUndo", () => {
    const tree = renderer.create(<IconUndo width={16} height={16} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
