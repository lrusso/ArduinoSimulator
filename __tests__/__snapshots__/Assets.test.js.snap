// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assets IconNew 1`] = `
<svg
  height={16}
  viewBox="0 0 1000 1000"
  width={16}
>
  <path
    d="M903 761l-142 142q-24 24 -64 40.5t-73 16.5h-480q-33 0 -56.5 -23.5t-23.5 -56.5v-864q0 -33 23.5 -56.5t56.5 -23.5h736q33 0 56.5 23.5t23.5 56.5v608q0 33 -16.5 73t-40.5 64zM858 715q3 -2 5 -5l4 -6h-163v163l6 -4t5 -5zM896 16q0 -7 -4.5 -11.5t-11.5 -4.5h-736 q-7 0 -11.5 4.5t-4.5 11.5v864q0 7 4.5 11.5t11.5 4.5h480q4 0 8 -0.5t8 -1.5v-254h254q1 -4 1.5 -8t0.5 -8v-608z"
    transform="translate(0 930) scale(-1,1) rotate(180)"
  />
</svg>
`;

exports[`Assets IconOpen 1`] = `
<svg
  height={16}
  viewBox="0 0 576 512"
  width={16}
>
  <path
    d="M572.694 292.093L500.27 416.248A63.997 63.997 0 0 1 444.989 448H45.025c-18.523 0-30.064-20.093-20.731-36.093l72.424-124.155A64 64 0 0 1 152 256h399.964c18.523 0 30.064 20.093 20.73 36.093zM152 224h328v-48c0-26.51-21.49-48-48-48H272l-64-64H48C21.49 64 0 85.49 0 112v278.046l69.077-118.418C86.214 242.25 117.989 224 152 224z"
    transform="translate(0 -60) scale(1,1.2)"
  />
</svg>
`;

exports[`Assets IconRedo 1`] = `
<svg
  height={16}
  viewBox="0 0 1000 1000"
  width={16}
>
  <path
    d="M576 712v248l384 -384l-384 -384v254q-168 4 -253.5 -46.5t-113.5 -127.5t-9 -168t62 -168q-105 114 -144.5 251.5t-4.5 255.5t148.5 196t314.5 73v0z"
    transform="translate(0 930) scale(-1,1) rotate(180)"
  />
</svg>
`;

exports[`Assets IconSave 1`] = `
<svg
  height={16}
  viewBox="0 0 1000 1000"
  width={16}
>
  <path
    d="M896 960h-896v-1024h1024v896zM512 832h128v-256h-128v256zM896 64h-768v768h64v-320h576v320h75l53 -53v-715z"
    transform="translate(-10 930) scale(-1,1) rotate(180)"
  />
</svg>
`;

exports[`Assets IconSearch 1`] = `
<svg
  height={16}
  viewBox="0 0 24 24"
  width={16}
>
  <path
    d="M11,5 C8.24078475,5 6,7.24078475 6,10 C6,12.7592153 8.24078475,15 11,15 C13.7592153,15 16,12.7592153 16,10 C16,7.24078475 13.7592153,5 11,5 Z M16.1517226,14.737509 L19.7071068,18.2928932 C20.0976311,18.6834175 20.0976311,19.3165825 19.7071068,19.7071068 C19.3165825,20.0976311 18.6834175,20.0976311 18.2928932,19.7071068 L14.5932864,16.0075 C13.5426554,16.6376337 12.3133603,17 11,17 C7.13621525,17 4,13.8637847 4,10 C4,6.13621525 7.13621525,3 11,3 C14.8637847,3 18,6.13621525 18,10 C18,11.826606 17.2990779,13.4906052 16.1517226,14.737509 Z"
    transform="translate(0 2)"
  />
</svg>
`;

exports[`Assets IconStart 1`] = `
<svg
  height={16}
  id="arduino_simulator_start_icon"
  viewBox="0 0 56.693 56.693"
  width={16}
>
  <path
    d="M46.279,26.484L20.113,7.48c-0.957-0.695-2.228-0.796-3.278-0.258c-1.056,0.537-1.721,1.622-1.721,2.807  v38.009c0,1.184,0.665,2.268,1.721,2.805c0.45,0.23,0.939,0.344,1.429,0.344c0.652,0,1.303-0.203,1.85-0.602l26.166-19.006  c0.818-0.594,1.298-1.539,1.298-2.548C47.577,28.023,47.094,27.077,46.279,26.484z"
  />
</svg>
`;

exports[`Assets IconStop 1`] = `
<svg
  height={16}
  id="arduino_simulator_stop_icon"
  viewBox="0 0 24 24"
  width={16}
>
  <path
    d="M16,6H8C6.9,6,6,6.9,6,8v8c0,1.1,0.9,2,2,2h8c1.1,0,2-0.9,2-2V8C18,6.9,17.1,6,16,6z"
    transform="translate(-0.5 -5) scale(1.1,1.5)"
  />
</svg>
`;

exports[`Assets IconUndo 1`] = `
<svg
  height={16}
  viewBox="0 0 1000 1000"
  width={16}
>
  <path
    d="M762 -64q43 77 62 168t-9 168t-113.5 127.5t-253.5 46.5v-254l-384 384l384 384v-248q201 5 314.5 -73t148.5 -196t-4.5 -255.5t-144.5 -251.5z"
    transform="translate(0 930) scale(-1,1) rotate(180)"
  />
</svg>
`;
